### PR TITLE
fcgene 1.0.7 (transfer from deprecated homebrew-science)

### PR DESCRIPTION
**Pull request by alexreg**

### CODE REVIEW

File: Formula/fcgene.rb [hunk 1]
```diff
@@ -0,0 +1,22 @@
+class Fcgene < Formula
```
Comment (fxcoudert) [inline Formula/fcgene.rb]: Missing description (`desc`)

**Review by fxcoudert — CHANGES_REQUESTED** [review]

File: Formula/fcgene.rb (hunk 1 above)
Comment (alexreg) [reply Formula/fcgene.rb]: Done.

File: Formula/fcgene.rb [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+                          "--disable-silent-rules",
+                          "--prefix=#{prefix}"
+    system "make"
+    system "make", "install"
+  end
+
+  test do
+    (testpath/"test.ped").write("1 1000000000 0 0 1 1 0 0 1 1\n1 1000000001 0 0 1 2 1 1 1 2\n")
+    (testpath/"test.map").write("1 rs0 0 1000\n1 rs10 0 1001\n")
+    system "#{bin}/fcgene", "--ped", "test.ped", "--map", "test.map"
```
Comment (fxcoudert) [inline Formula/fcgene.rb]: Can we test the output? Or that a file is written?

Comment (alexreg) [reply Formula/fcgene.rb]: We can test if the log file is generated. Is that worth doing?

**Review by fxcoudert — APPROVED** [review]

File: Formula/fcgene.rb [hunk 3]
```diff
@@ -0,0 +1,23 @@
+class Fcgene < Formula
+  desc "Format-converting tool for genotype data"
+  homepage "https://sourceforge.net/projects/fcgene/"
+  url "https://downloads.sourceforge.net/project/fcgene/fcgene-1.0.7.tar.gz"
+  sha256 "4e1f85f2ec812e2528bd19b6c18ecf297666cd83046e003bc57d9ed5f25783d6"
+
+  def install
+    ENV["AC_CXX"] = ENV["CXX"] # otherwise CXX is ignored
```
Comment (ilovezfs) [inline Formula/fcgene.rb]: The build succeeds without this line so it should be removed.

File: Formula/fcgene.rb [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+    system "./configure", "--disable-debug",
+                          "--disable-dependency-tracking",
+                          "--disable-silent-rules",
+                          "--prefix=#{prefix}"
+    system "make"
+    system "make", "install"
+  end
+
+  test do
+    (testpath/"test.ped").write("1 1000000000 0 0 1 1 0 0 1 1\n1 1000000001 0 0 1 2 1 1 1 2\n")
```
Comment (ilovezfs) [inline Formula/fcgene.rb]: This should use a heredoc.

File: Formula/fcgene.rb [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+    (testpath/"test.map").write("1 rs0 0 1000\n1 rs10 0 1001\n")
```
Comment (ilovezfs) [inline Formula/fcgene.rb]: This should use a heredoc.

File: Formula/fcgene.rb [hunk 6]
```diff
@@ -0,0 +1,23 @@
+class Fcgene < Formula
+  desc "Format-converting tool for genotype data"
+  homepage "https://sourceforge.net/projects/fcgene/"
```
Comment (ilovezfs) [inline Formula/fcgene.rb]: The homepage seems to actually be https://www.imise.uni-leipzig.de/en/Groups/GenStat/Tools/index.jsp

**Review by ilovezfs — CHANGES_REQUESTED** [review]

File: Formula/fcgene.rb (hunk 3 above)
Comment (ilovezfs) [reply Formula/fcgene.rb]: No, there are compiler shims that handle that.

Comment (alexreg) [reply Formula/fcgene.rb]: Got it.

**Comment by ilovezfs** [comment]
Actually, this is going to be a better fit for a [personal tap](https://github.com/Homebrew/brew/blob/master/docs/How-to-Create-and-Maintain-a-Tap.md#how-to-create-and-maintain-a-tap). It didn't qualify for migration based on analytics data when we were migrating formulae over from science and there hasn't been a release in 3 years. Thanks for the PR though @alexreg

**Comment by alexreg** [comment]
Okay, I see...

**Pull request closed without merging**